test/Ocelot.AcceptanceTests/ServiceDiscovery/KubernetesServiceDiscoveryTests.cs

```diff
@@ -311,15 +309,24 @@ private void GivenThereIsAFakeKubernetesProvider(EndpointsV1 endpoints, bool isS
     }
 
     private void WithKubernetes(IServiceCollection services) => services
-        .AddOcelot().AddKubernetes()
-        .Services.RemoveAll<IKubeApiClient>().AddSingleton(_clientFactory);
+        .AddOcelot().AddKubernetes(false)
+        .Services.Configure(_kubeClientOptionsConfigure)
+        .Replace(GetValidateScopesDescriptor());
 
     private void WithKubernetesAndRoundRobin(IServiceCollection services) => services
-        .AddOcelot().AddKubernetes()
+        .AddOcelot().AddKubernetes(false)
         .AddCustomLoadBalancer<RoundRobinAnalyzer>(GetRoundRobinAnalyzer)
         .Services
-        .RemoveAll<IKubeApiClient>().AddSingleton(_clientFactory)
-        .RemoveAll<IKubeServiceCreator>().AddSingleton<IKubeServiceCreator, FakeKubeServiceCreator>();
+        .Configure(_kubeClientOptionsConfigure)
+        .RemoveAll<IKubeServiceCreator>().AddSingleton<IKubeServiceCreator, FakeKubeServiceCreator>()
+        .Replace(GetValidateScopesDescriptor());
```

Comment:
It appears that the `Configure` and `Replace` methods ought to be positioned more closely together.
```suggestion
        .Configure(_kubeClientOptionsConfigure)
        .Replace(GetValidateScopesDescriptor());
        .RemoveAll<IKubeServiceCreator>().AddSingleton<IKubeServiceCreator, FakeKubeServiceCreator>()
```